# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/renderer/shade.rs
# Commit: c0f58fd74e56
# Full Hash: c0f58fd74e56b24f67fccae08097d151e21a0f80
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2024-05-09 16:15:25
# Regressor Bug: 1824502
# File Overlap Count: 1
# Description:
#   Bug 1824502 - Implement native SVG filter rendering in WebRender r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D209646
# ==============================================================================

diff -r 3e2ec3a3b738 -r c0f58fd74e56 gfx/wr/webrender/src/renderer/shade.rs
--- a/gfx/wr/webrender/src/renderer/shade.rs	Thu May 09 02:12:39 2024 +0000
+++ b/gfx/wr/webrender/src/renderer/shade.rs	Thu May 09 04:09:00 2024 +0000
@@ -263,6 +263,7 @@
                 VertexArrayKind::Scale => &desc::SCALE,
                 VertexArrayKind::Resolve => &desc::RESOLVE,
                 VertexArrayKind::SvgFilter => &desc::SVG_FILTER,
+                VertexArrayKind::SvgFilterNode => &desc::SVG_FILTER_NODE,
                 VertexArrayKind::Composite => &desc::COMPOSITE,
                 VertexArrayKind::Clear => &desc::CLEAR,
                 VertexArrayKind::Copy => &desc::COPY,
@@ -601,6 +602,7 @@
     pub cs_radial_gradient: LazilyCompiledShader,
     pub cs_conic_gradient: LazilyCompiledShader,
     pub cs_svg_filter: LazilyCompiledShader,
+    pub cs_svg_filter_node: LazilyCompiledShader,
 
     // Brush shaders
     brush_solid: BrushShader,
@@ -770,6 +772,16 @@
             profile,
         )?;
 
+        let cs_svg_filter_node = LazilyCompiledShader::new(
+            ShaderKind::Cache(VertexArrayKind::SvgFilterNode),
+            "cs_svg_filter_node",
+            &[],
+            device,
+            options.precache_flags,
+            &shader_list,
+            profile,
+        )?;
+
         let ps_mask = LazilyCompiledShader::new(
             ShaderKind::Cache(VertexArrayKind::Mask),
             "ps_quad_mask",
@@ -1129,6 +1141,7 @@
             cs_border_solid,
             cs_scale,
             cs_svg_filter,
+            cs_svg_filter_node,
             brush_solid,
             brush_image,
             brush_fast_image,
@@ -1300,6 +1313,7 @@
         self.cs_blur_a8.deinit(device);
         self.cs_blur_rgba8.deinit(device);
         self.cs_svg_filter.deinit(device);
+        self.cs_svg_filter_node.deinit(device);
         self.brush_solid.deinit(device);
         self.brush_blend.deinit(device);
         self.brush_mix_blend.deinit(device);